### PR TITLE
feat(Search samples) Modify samples endpoint to allow for searching.

### PR DESCRIPTION
**Pull request by Vince-janv**
### Description
Adds a sample search functionality.

### Added
- Function for querying samples based on substring
### Changed
- Added a enquiry parameter to the samples endpoint to allow for sample searches

**Steps to consider while deploying**
- Configuration changes: N/A
- Documentation updates: N/A
- Inform users by email: N/A

### Review:
- [ ] Code approved by
- [ ] Tests executed on stage by:  (Document the test done with screen shots and description.)
- [ ] "Merge and deploy" approved by

This [version](https://semver.org/) is a:
- [ ] **MAJOR** - when you make incompatible API changes
- [x] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

**Comment by Vince-janv** [comment]
### Test on the genotype stage swagger:
Screenshot 2022-09-16 at 11 03 53

**Review by elevu — APPROVED** [review]
Approving the PR, just some tiny comments :)
- The endpoints returns 500 when making some calls like `https://genotype-stage.scilifelab.se/api/v2/samples/?skip=0&limit=2&enquiry=ACC&plate_id=ID127&incomplete=true&commented=true&status_missing=true` but I guess it is not relevant to this PR.
- The search by plate id seems also not to work, but maybe I am wrong and just searching for non existing combination (commented true and so on).
- No tests? :D

File: genotype_api/crud/samples.py [hunk 1]
```diff
@@ -17,6 +17,11 @@ def get_sample(session: Session, sample_id: str) -> Sample:
     return session.exec(statement).one()
 
 
+def get_samples(statement: SelectOfScalar, enquiry: str) -> SelectOfScalar:
+    """Gets samples where the sample id contains the enquiry."""
+    return statement.where(Sample.id.contains(enquiry))
```
[the remark this answers is not in the document]
Comment (Vince-janv) [reply genotype_api/crud/samples.py:22]: Good point 👍

**Comment by Vince-janv** [comment]
> Approving the PR, just some tiny comments :)
> 
> * The endpoints returns 500 when making some calls like `https://genotype-stage.scilifelab.se/api/v2/samples/?skip=0&limit=2&enquiry=ACC&plate_id=ID127&incomplete=true&commented=true&status_missing=true` but I guess it is not relevant to this PR.
> * The search by plate id seems also not to work, but maybe I am wrong and just searching for non existing combination (commented true and so on).
> * No tests? :D

Thanks for the review @elevu! 😊 

- I played around a bit with the swagger UI and it seems that `incomplete=true` in combination with some other options returns 500. Probably not related to anything I changed, but I can take a look.
- hmm, plate id works for me 🤔 
- That's why I wanted your review. How would you suggest to test these kind of PRs except via the swagger UI. There are no pytests in the genotype-api repo 😨

File: genotype_api/api/endpoints/samples.py [hunk 2]
```diff
@@ -86,14 +87,18 @@ def read_sample(
 def read_samples(
     skip: int = 0,
     limit: int = Query(default=10, lte=10),
+    enquiry: Optional[str] = None,
```
Comment (henrikstranneheim) [inline genotype_api/api/endpoints/samples.py:90]:
```suggestion
    sample_id: Optional[str] = None,
```

File: genotype_api/crud/samples.py [hunk 3]
```diff
@@ -17,6 +17,11 @@ def get_sample(session: Session, sample_id: str) -> Sample:
     return session.exec(statement).one()
 
 
+def get_samples(statement: SelectOfScalar, enquiry: str) -> SelectOfScalar:
+    """Gets samples where the sample id contains the enquiry."""
```
Comment (henrikstranneheim) [inline genotype_api/crud/samples.py:21]:
```suggestion
    """Return samples containing sample id."""
```

File: genotype_api/crud/samples.py [hunk 4]
```diff
@@ -17,6 +17,11 @@ def get_sample(session: Session, sample_id: str) -> Sample:
...
+    return statement.where(Sample.id.contains(enquiry))
+
+
 def create_sample(session: Session, sample: Sample) -> Sample:
     """Adding a sample to db"""
```
Comment (henrikstranneheim) [inline genotype_api/crud/samples.py:26]:
```suggestion
    """Creating a sample in the database."""
```

**Review by henrikstranneheim — APPROVED** [review]
👍

**Comment by henrikstranneheim** [comment]
@Vince-janv For API endpoint testing there is a framework in fastAPI to use. See https://github.com/Clinical-Genomics/atlas/tree/master/tests/api for inspiration

**Comment by Vince-janv** [comment]
@henrikstranneheim Should I spend time starting to write tests here or should I merge this and prioritise something else?

**Comment by henrikstranneheim** [comment]
@Vince-janv I think it is no worse then when you arrived. I think we should not spend time on this technical debt now, as this code seem to be of poor standards and would require some work. We should make a project out of it and work on it in a team in a more focused way.

**Pull request merged**